File: internships/tinkoff/summer2023/4.py
# переделать


n = int(input())
arr = list(map(int, input().split()))
d = {}
mx = 1

if len(set(arr)) == 1:
    print(len(arr))
else:
    for i in range(n):
        if arr[i] in d.keys():
            d[arr[i]] += 1
        else:
            d[arr[i]] = 1

        d2 = {}

        for j in d.values():
            if j in d2.keys():
                d2[j] += 1
            else:
                d2[j] = 1

        ln = len(d2.keys())
        if ln == 1:
            if list(d2.values())[0] > 1:
                mx = i+1
        elif ln == 2:
            a, b = list(d2.values())
            if 1 in d2.values():
                mx = i+1

    print(mx)

File: internships/yandex/mobile/mobile_summer_2023/tasks/4.swift
var k = Int(readLine()!)!
var n = Int(readLine()!)!
let input = readLine() ?? ""
let arr = input.split(separator: " ").compactMap { Int($0)! }

var minCount = Int.max

func pow_(_ a: Int, _ b: Int) -> Int {
    var result = 1
    for _ in 0..<b {
        result *= a
    }
    return result
}

for i in 0..<n {
    if arr[i] == 0 {
        minCount = 0
        break
    }

    let count = pow_(2, i)
    let div = k % count == 0 ? Int(Float(k) / Float(count)) : Int(Float(k) / Float(count)) + 1
    let sum_ = div * arr[i]
    
    if sum_ < minCount {
        minCount = sum_
    }
}

print(minCount)